src/expr.rs

```diff
@@ -1857,14 +1857,14 @@ pub(crate) fn rewrite_assign_rhs<S: Into<String>, R: Rewrite>(
     rewrite_assign_rhs_with(context, lhs, ex, shape, RhsTactics::Default)
 }
 
-pub(crate) fn rewrite_assign_rhs_with<S: Into<String>, R: Rewrite>(
+pub(crate) fn rewrite_assign_rhs_node<R: Rewrite>(
     context: &RewriteContext<'_>,
-    lhs: S,
+    lhs: &str,
     ex: &R,
     shape: Shape,
     rhs_tactics: RhsTactics,
+    has_impl: bool,
```

Comment:
Hmm, adding a logic specific for handling `impl` to `rewrite_assign` does not seem a good idea to me.

Reply:
I also think to use a wrapper GenericBounds and implement Rewrite trait is better than my idea. Thanks!